Sources/Client/Client/Client.swift

```diff
@@ -133,6 +133,11 @@ public final class Client: Uploader {
         self.onUserUpdateObservers.values.forEach({ $0(newUser) })
     }
     
+    #if DEBUG
+    /// Called when a new outgoing event is about to be sent. Meant to be used only for testing purposes.
+    var outgoingEventsTestLogger: ((EventType) -> Void)?
+    #endif
```

Comment:
This smells bad and looks like a hardcoded solution, but it's really nice for test usage.

Reply:
You must play with the cards you have 🤷‍♂️ 😄